### PR TITLE
V0.9.0

### PR DESCRIPTION
**Pull request by esc**
Prepping the v0.9.0 release.

### CODE REVIEW

**Comment by esc** [comment]
* Tests pass on my machine (works for me)
 * Tests fail on my AWS Windows.

**Comment by esc** [comment]
grml :/

**Comment by esc** [comment]
PS C:\Users\Administrator\git\bcolz> python -c "import bcolz ; bcolz.test(heavy=True)"
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
bcolz version:     0.9.0
NumPy version:     1.9.1
Blosc version:     1.4.1 ($Date:: 2014-07-08 #$)
Blosc compressors: ['blosclz', 'lz4', 'lz4hc', 'snappy', 'zlib']
Numexpr version:   2.3.1
Python version:    2.7.8 |Continuum Analytics, Inc.| (default, Jul  2 2014, 15:12:11) [MSC v.1500 64 bit (AMD64)]
Byte-ordering:     little
Detected cores:    1
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Performing the complete test suite!
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
......s......s......s......s............................................................................................
....................................................................................................................s...
........................FFF.............................................................................................
...................................E..............................ss....................................................
........................................................................................................................
........................................................................................................................
........................................................................................................................
..............................................................
======================================================================
ERROR: test_add_new_column_ondisk_other_carray_rootdir (test_ctable.add_del_colDiskTest)
Testing addcol with different rootdir.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Administrator\git\bcolz\bcolz\tests\test_ctable.py", line 437, in test_add_new_column_ondisk_other_carr
ay_rootdir
    os.remove(fn)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\adm
ini~1\\appdata\\local\\temp\\2\\tmpzbzpj1'

======================================================================
FAIL: test_repr_disk_array_append (test_carray.reprDiskTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Administrator\git\bcolz\bcolz\tests\test_carray.py", line 2148, in test_repr_disk_array_append
    self.assertEqual(expected, repr(y))
AssertionError: "carray((0,), float64)\n  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00\n  cparams := cparams(clevel=5, shuff
le=True, cname='blosclz')\n  rootdir := 'c:\\users\\admini~1\\appdata\\local\\temp\\2\\bcolz-reprDiskTest4yueru'\n  mode
    := 'a'\n[]" != "carray((0L,), float64)\n  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00\n  cparams := cparams(clevel=5, s
huffle=True, cname='blosclz')\n  rootdir := 'c:\\users\\admini~1\\appdata\\local\\temp\\2\\bcolz-reprDiskTest4yueru'\n
mode    := 'a'\n[]"

======================================================================
FAIL: test_repr_disk_array_read (test_carray.reprDiskTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Administrator\git\bcolz\bcolz\tests\test_carray.py", line 2142, in test_repr_disk_array_read
    self.assertEqual(expected, repr(x))
AssertionError: "carray((0,), float64)\n  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00\n  cparams := cparams(clevel=5, shuff
le=True, cname='blosclz')\n  rootdir := 'c:\\users\\admini~1\\appdata\\local\\temp\\2\\bcolz-reprDiskTestzou8qb'\n  mode
    := 'r'\n[]" != "carray((0L,), float64)\n  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00\n  cparams := cparams(clevel=5, s
huffle=True, cname='blosclz')\n  rootdir := 'c:\\users\\admini~1\\appdata\\local\\temp\\2\\bcolz-reprDiskTestzou8qb'\n
mode    := 'r'\n[]"

======================================================================
FAIL: test_repr_disk_array_write (test_carray.reprDiskTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Administrator\git\bcolz\bcolz\tests\test_carray.py", line 2137, in test_repr_disk_array_write
    self.assertEqual(expected, repr(x))
AssertionError: "carray((0,), float64)\n  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00\n  cparams := cparams(clevel=5, shuff
le=True, cname='blosclz')\n  rootdir := 'c:\\users\\admini~1\\appdata\\local\\temp\\2\\bcolz-reprDiskTestk1ve14'\n  mode
    := 'w'\n[]" != "carray((0L,), float64)\n  nbytes: 0; cbytes: 16.00 KB; ratio: 0.00\n  cparams := cparams(clevel=5, s
huffle=True, cname='blosclz')\n  rootdir := 'c:\\users\\admini~1\\appdata\\local\\temp\\2\\bcolz-reprDiskTestk1ve14'\n
mode    := 'w'\n[]"

----------------------------------------------------------------------
Ran 902 tests in 40.687s

FAILED (failures=3, errors=1, skipped=7)

**Comment by esc** [comment]
The repr tests fails due to ``0`` != ``0L`` not sure about the file lock..

**Comment by esc** [comment]
I think I fixed the creation of temporary files (still the directory lingers after the test, suboptimal)

**Comment by esc** [comment]
The problem is, that on windows:

```
    property len:
        "The length (leading dimension) of this object."
        def __get__(self):
            if self._dtype.char == 'O':
                return len(self.chunks)
            else:
                # Important to do the cast in order to get a npy_intp result
                return <npy_intp> cython.cdiv(self._nbytes, self.atomsize)
```

Here the last line returns a long into python space.. on linux an int..

**Comment by esc** [comment]
We can't remove the cast either, because then the tests fail...

**Comment by esc** [comment]
Added a bit of a hacky fix...

**Comment by esc** [comment]
I squeezed #178 in too.

**Comment by esc** [comment]
test suite now passes on windows.

**Comment by esc** [comment]
@alimanfoo I mentioned scikit-allel here, let me know if you would like a different text.

**Comment by esc** [comment]
@FrancescAlted I think it's ready to ship, could you glance over the ANNOUNCE.rst one last time, please.

**Comment by FrancescAlted** [comment]
Uh, the 'What's new' section is still mostly empty.  Am I missing something?

**Comment by esc** [comment]
Not sure, maybe look at:

https://github.com/Blosc/bcolz/pull/177/files

File: ANNOUNCE.rst [hunk 1]
```diff
@@ -5,7 +5,34 @@ Announcing bcolz 0.9.0
...
+unexpected behaviour, copying also copies the Python object, whereas a
+moving the data doesn't. When removing columns with 'delcol' you can
+specify the keyword argument 'keep' if you wish to keep the data in case
+of disk-backed carrays. Additionally, ctable now supports an 'auto_flush'
+keyword that makes it flush to disk automatically after any methods that
+may write data.
+
+Another important aspect was how the GIL is handled. In this release, we
+do keep the GIL while calling Blosc compress and decompress in order to
+support bug-free operation of newer Blosc versions that no longer have a
```
Comment (FrancescAlted) [inline ANNOUNCE.rst]: 'bug-free'  -> 'lock-free'

File: ANNOUNCE.rst [hunk 2]
```diff
@@ -5,7 +5,34 @@ Announcing bcolz 0.9.0
...
+global state.
+
+Furthermore we now distribute the 'carray_ext.pxd' as part of  the
+package via PyPi to ease building applications on bcolz, for example
+*bquery*.
+
+Finally, some technical debt has been removed and the Sphinx based API
```
Comment (FrancescAlted) [inline ANNOUNCE.rst]: Finally, the Sphinx based API is now autogenerated...

File: ANNOUNCE.rst [hunk 3]
```diff
@@ -5,7 +5,34 @@ Announcing bcolz 0.9.0
...
+tables.  When adding a disk-backed carray to a ctable, you can now use
+the boolean keyword argument 'move' to either move the data (True) or
+copy it (False) into the directory of the original ctable. To avoid
+unexpected behaviour, copying also copies the Python object, whereas a
+moving the data doesn't. When removing columns with 'delcol' you can
+specify the keyword argument 'keep' if you wish to keep the data in case
+of disk-backed carrays. Additionally, ctable now supports an 'auto_flush'
+keyword that makes it flush to disk automatically after any methods that
+may write data.
+
```
Comment (FrancescAlted) [inline ANNOUNCE.rst]: The sentence above is good but this is an announcement, and as such, I tend to make a shorter summary and then redirect to the release notes for people interested in more details.

File: ANNOUNCE.rst (hunk 1 above)
Comment (esc) [inline ANNOUNCE.rst]: done

File: ANNOUNCE.rst (hunk 2 above)
Comment (esc) [inline ANNOUNCE.rst]: done

File: ANNOUNCE.rst (hunk 3 above)
Comment (esc) [inline ANNOUNCE.rst]: I have shortened it a little.

**Comment by alimanfoo** [comment]
Sorry I didn't reply to this at the time, thanks for the mention, text is
fine.

On Saturday, May 2, 2015, Valentin Haenel <notifications@github.com> wrote:

> @alimanfoo <https://github.com/alimanfoo> I mentioned scikit-allel here,
> let me know if you would like a different text.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Blosc/bcolz/pull/177#issuecomment-98352238>.
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Web: http://purl.org/net/aliman
Email: alimanfoo@gmail.com
Tel: +44 (0)1865 287721

**Pull request merged**